docs/cpp/customize-cpp-settings.md

```diff
@@ -0,0 +1,174 @@
+
+### Recursive includes properties
+
+- `reduce`
+  When a recursive `includePath` entry is expanded, it can result in a very large set of include paths for IntelliSense to process when resolving `#include` statements in your source files. Sending a large set of include paths to the IntelliSense compiler can have an impact on the performance of IntelliSense on some systems. By default, the extension will reduce the set of include paths to the smallest possible set by first tag parsing the source files to search for `#include` statements and determining which include paths are needed. This reduction process is the same behavior as the `always` option for this setting. This behavior trades off some initial overhead so that IntelliSense can potentially be faster later on. Setting this property to `never` will provide the full recursive expansion of include paths to the IntelliSense process. By not parsing any files up front, this behavior trades off potential performance later on to ensure IntelliSense can start up more quickly when source files are opened. In general, reducing the number of recursive include paths in your configuration might improve IntelliSense performance when a large number of paths are involved.
+
+- `priority`
+  The priority of recursive include path searches when resolving `#include` statements. If set to `beforeSystemIncludes`, the recursive include paths are searched before system include paths. If set to `afterSystemIncludes`, the recursive include paths will be searched after system include paths. `beforeSystemIncludes` would more closely reflect the search order of a compiler leading to more predictability, while `afterSystemIncludes` might result in improved performance.
+
+- `order`
+  Whether subdirectories of recursive includes are searched `breadthFirst` or `depthFirst`.
+
+## Supported variables
+
+You can allow `tasks.json` or `launch.json` to query the current active configuration from `c_cpp_properties.json`. To do this, use the variable `${command:cpptools.activeConfigName}` as an argument in a `tasks.json` or `launch.json` script.
+
+### Default VS Code Settings
+
+All default VS Code settings, such as C_Cpp.default.includePath, are supported in c_cpp_properties.json. The only exception is
+
+```json
+C_Cpp.default.systemIncludePath : string[]
+```
+
+This setting allows you to specify the system include path separately from the include path. However, the selected system include path the C++ extension receives from the compiler is not passed to the IntelliSense process. This is only used in rare scenarios since it overwrites the standard compiler behavior, for example, if your compiler is not supported. Instead, use the setting `compilerArgs` and using the "-isystem" flag to specify system headers, which is a better solution in most scenarios.
```

Comment:
```suggestion
This setting allows you to specify the system include path separately from the include path. However, the selected system include path that the C++ extension receives from the compiler is not passed to the IntelliSense process. This is only used in rare scenarios since it overwrites the standard compiler behavior, for example, if your compiler is not supported. Instead, use the setting `compilerArgs` and using the "-isystem" flag to specify system headers, which is a better solution in most scenarios.
```